azurerm/internal/services/network/express_route_port_resource.go

```diff
@@ -0,0 +1,514 @@
+				Computed: true,
+			},
+			"interface_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"patch_panel_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"rack_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"connector_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	},
+}
+
+func resourceArmExpressRoutePort() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmExpressRoutePortCreateUpdate,
+		Read:   resourceArmExpressRoutePortRead,
+		Update: resourceArmExpressRoutePortCreateUpdate,
+		Delete: resourceArmExpressRoutePortDelete,
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.ExpressRoutePortID(id)
+			return err
+		}),
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.ExpressRoutePortName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"location": location.Schema(),
+
+			"peering_location": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"bandwidth_in_gbps": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.IntAtLeast(1),
+			},
+
+			"encapsulation": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(network.Dot1Q),
+					string(network.QinQ),
+				}, false),
+			},
+
+			"identity": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"identity_ids": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MinItems: 1,
+							Elem: &schema.Schema{
+								Type:             schema.TypeString,
+								ValidateFunc:     validation.NoZeroValues,
+								DiffSuppressFunc: suppress.CaseDifference,
+							},
+						},
+
+						"type": {
+							Type:     schema.TypeString,
+							Required: true,
+							// TODO: The "ignoreCase" and diff suppression function can be removed once
+							// following issue get resolved:
+							// https://github.com/Azure/azure-rest-api-specs/issues/12330
+							ValidateFunc: validation.StringInSlice([]string{
+								string(network.ResourceIdentityTypeUserAssigned),
+							}, true),
+							DiffSuppressFunc: suppress.CaseDifference,
+						},
+					},
+				},
+			},
+
+			"link1": expressRoutePortSchema,
+
+			"link2": expressRoutePortSchema,
```

Comment:
could we use a list here so link.0/link.1 works instead?

Reply:
We can make it to be a list with exactly two items, whilst it would require users to set the first link in order to update anything on the second link. Otherwise, we can allow users to modify each link independently.
Also confirmed from service team that there will be no more links added for the express route port resource, only twos for high availability.

Reply:
thats a good point, lets keep it like this then